### PR TITLE
German translation of "Intro to Storybook" (React)

### PR DESCRIPTION
**Pull request by renet**
This is the German translation of the "Intro to Storybook" for React. #3

- Its been updated to reflect today's latest changes.
- It's not been reviewed by another native speaker, yet. Any volunteers?

### CODE REVIEW

**Comment by domyen** [comment]
This is amazing @renet! Thank you 🙏  
I will socialize it with the community to get some proofreaders :)

**Comment by jonniebigodes** [comment]
@renet saw that @domyen reached out to the community and i'm also doing the same to help this get proofread and the translation added

File: src/lib/getLanguageName.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 const languageNameMap = {
+  de: 'Deutsch',
```
Comment (domyen) [inline src/lib/getLanguageName.js]: Can we move this to the end of the array? The language order is intentional and displayed in the UI.

Comment (renet) [reply src/lib/getLanguageName.js]: Sure, I thought the order was alphabetical, a closer look would have prevented that. Will change it soon.

File: content/intro-to-storybook/react/de/composite-component.md [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+tocTitle: 'Komposition'
+description: 'Setze eine Komposition aus einfachen Komponenten zusammen'
+commit: '8db511e'
+---
+
+Im letzten Kapitel haben wir unsere erste Komponente entwickelt; in diesem Kapitel erweitern wir das Gelernte, um `TaskList` zu bauen, eine Liste von Aufgaben. Lass uns Komponenten miteinander kombinieren und sehen, was passiert, wenn mehr Komplexität ins Spiel kommt.
+
+## Liste von Aufgaben
+
+Taskbox hebt eine angeheftete Aufgabe hervor, indem es sie über anderen Aufgaben positioniert. Hieraus ergeben sich zwei Varianten der `TaskList`, für die wir Stories anlegen müssen: Normale Aufgaben sowie normale und angeheftete Aufgaben.
```
[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/composite-component.md]: Ist aber beabsichtigt. Es geht darum, dass in einer Liste nur die normalen Aufgaben, in der anderen Liste aber "normale und angeheftete" Aufgaben gemeinsam enthalten sind. Ich überlege noch mal, ob es evtl. eine eindeutigere Formulierung dafür gibt - wenn dir etwas Gutes einfällt, immer her damit. :)

File: content/intro-to-storybook/react/de/get-started.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+commit: ebe2ae2
+---
+
+Storybook läuft parallel zu deiner App im Entwicklungs-Modus. Es hilft dir, UI-Komponenten unabhängig von Business-Logik und vom Kontext deiner App zu entwickeln. Diese Edition von "Lerne Storybook" ist für React; Es gibt andere Editionen für [Vue](/vue/de/get-started) und [Angular](/angular/de/get-started).
+
+![Storybook und deine App](/intro-to-storybook/storybook-relationship.jpg)
+
+## React Storybook einrichten
+
+Es sind einige Schritte nötig, um den Build Prozess in deiner Umgebung einzurichten. Lass uns zu Beginn [Create React App](https://github.com/facebook/create-react-app) (CRA) nutzen, um dein Build System einzurichten und [Storybook](https://storybook.js.org/) und [Jest](https://facebook.github.io/jest/) Tests in der von dir erstellten App zu ermöglichen. Dazu führen wir folgende Befehle aus:
```
[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/get-started.md]: The actual meaning of the words are not 100% clear to me here. Maybe what it means is "enable Storybook tests (1) and Jest tests (2)", or it means "Storybook (1) and Jest tests (2)".  I assume the latter. Maybe @domyen or @jonniebigodes can help us here. Does this paragraph refer to "Storybook tests" or Storybook itself?

File: content/intro-to-storybook/react/de/get-started.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+![Taskbox UI](/intro-to-storybook/ss-browserchrome-taskbox-learnstorybook.png)
+
+<div class="aside">
+Wenn du das Styling anpassen willst, findest du die LESS Quelldateien im GitGub Repo.
+</div>
+
+## Assets hinzufügen
+
+Wir müssen noch die font und icon [Verzeichnisse](https://github.com/chromaui/learnstorybook-code/tree/master/public) im `public/` Ordner hinzufügen. Nachdem wir das Styling und die Assets hinzugefügt haben, sieht die App noch ein bisschen seltsam aus. Das ist OK. Noch müssen wir nicht an der App arbeiten. Wir legen jetzt los und bauen unsere erste Komponente!
```
[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/get-started.md]: Da `public/`einen Schrägstrich enthält, habe ich die Reihenfolge einfach geändert in "Ordner `public/`". Ist das okay?

File: content/intro-to-storybook/react/de/test.md [hunk 5]
```diff
@@ -0,0 +1,129 @@
+---
+title: 'UI-Komponenten testen'
+tocTitle: 'Tests'
+description: 'Lerne, wie man UI-Komponenten testen kann'
+commit: 78a45d1
+---
+
+Kein Storybook Tutorial ist komplett ohne Tests. Tests sind essenziel für die Erstellung hochwertiger UIs. In modularen Systemen können auch kleinste Änderungen einen langen Rattenschwanz nach sich ziehen. Bisher sind uns drei Arten von Tests begegnet:
```
[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/test.md]: Habe es in `essenziell` korrigiert, ist für mich die gewohntere Schreibweise, als mit `t`, und dennoch zulässig. Aber das `l` hat gefehlt. ;)

File: content/intro-to-storybook/react/de/test.md [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+    src="/intro-to-storybook/website-workflow-review-merge-optimized.mp4"
+    type="video/mp4"
+  />
+</video>
+
+Da sich moderne Apps aus Komponenten zusammensetzen, ist es wichtig, dass wir auf Komponenten-Ebene testen. Das hilft uns dabei, den Ursprung eines Problems leichter zu identifizieren, anstatt auf seine Symptome in Screens und Kompositionen zu reagieren.
+
+## Änderungen übernehmen
+
+Sobald wir mit der Auswertung fertig sind, können wir die Änderungen in der UI mit gutem Gefühl übernehmen -- uns gewiss, dass die Änderungen keine unbeabsichtigten Bugs verursachen. Wenn dir der neue rote Hintergrund gefällt, akzeptiere die Änderungen. Andernfalls gehe zum vorherigen Zustand zurück.
```
[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/test.md]: War kein Typo, habe es umformuliert.

File: content/intro-to-storybook/react/de/data.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+    tasks: tasks.filter(t => t.state === 'TASK_INBOX' || t.state === 'TASK_PINNED'),
+  }),
+  dispatch => ({
+    onArchiveTask: id => dispatch(archiveTask(id)),
+    onPinTask: id => dispatch(pinTask(id)),
+  })
+)(PureTaskList);
+```
+
+Ab jetzt werden unsere Storybook Tests fehlschlagen, da die `TaskList` jetzt ein Container ist und keine Props mehr erwartet. Stattdessen verbindet es sich mit dem Store und übergibt Props an die `PureTaskList` Komponente, die es umschließt.
```
[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/data.md]: Habe in `er` geändert, da Bezug auf Container.

[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/data.md]: s.o.

File: content/intro-to-storybook/react/de/get-started.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (domyen) [reply content/intro-to-storybook/react/de/get-started.md]: @renet It's the latter – sorry for the confusion!

[the remark this answers is not in the document]
Comment (domyen) [reply content/intro-to-storybook/react/de/get-started.md]: It's the latter

[the remark this answers is not in the document]
Comment (jonniebigodes) [reply content/intro-to-storybook/react/de/get-started.md]: @renet and @jonathanherdt first and foremost let me apologize for my lack of knowledge of the German language. Now onto the question, if i'm reading this correctly it's refering to the first paragraph of the Get Started section correct? When i was working on the translation i asked myself the same question. I read the paragraph and checked the rest of the documentation. The intention in that paragraph is to tell the reader that he/she will use CRA for building the app and both Jest and Storybook for testing and that's it. At this stage he/she doesn't know what Storybook is and how to use it and depending on the case he/she might not know what Jest is also. That will be the focus of the tutorial. Go over these toolings and how they can be used first, first separately and then used together. So going with 

> Storybook (1) and Jest tests (2)

seems to me that is more accurate. @domyen do you agree?

[the remark this answers is not in the document]
Comment (domyen) [reply content/intro-to-storybook/react/de/get-started.md]: Yea that's right @jonniebigodes. The English version could be more clear in this regard.

[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/get-started.md]: Well, I don't think it's necessary to be more precise about this in this part of the tutorial. But translating it to German in a proper way requires additional information. Thank you for clarification! :)

File: content/intro-to-storybook/react/de/using-addons.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+description: 'Lerne an einem bekannten Beispiel, Addons zu integrieren und zu nutzen'
+commit: 'dac373a'
+---
+
+Storybook rühmt sich eines robuten [Addon-Systems](https://storybook.js.org/addons/introduction/), über das sich die Entwicklungserfahrung all deiner Teammitglieder verbessern lässt. Wenn du diesem Tutorial linear gefolgt bist, haben wir bereits einige Addons erwähnt und du hast bereits eines im [Kapitel über Tests](/react/de/test/) implementiert.
+
+<div class="aside">
+<strong>Auf der Suche nach einer Liste verfügbarer Addons?</strong>
+<br/>
+😍 <a href="https://storybook.js.org/addons/addon-gallery/">Hier</a> findest du die Liste offiziell unterstützter und aktiv unterstützer Community Addons.
```
[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/using-addons.md]: Würde "... offiziell unterstützter und von der Community aktiv unterstützter Addons." schreiben. Wäre das ok?

File: content/intro-to-storybook/react/de/using-addons.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+</div>
+
+Wir könnten unendlich viel über die Verwendung von Addons für all deine speziellen Anwendungsfälle schreiben. Für's Erste, lass uns auf die Integration eines der am weitesten verbreiteten Addons innerhalb des Storybook-Ökosystems hinarbeiten: [knobs](https://github.com/storybooks/storybook/tree/master/addons/knobs).
+
+## Knobs einrichten
+
+Knobs (Knöpfe) sind eine tolle Möglichkeit für Designer und Entwickler, um in einer kontrollierten Umgebung mit Komponenten herum zu experimentieren und zu spielen, ohne die Notwendigkeit von Code! Im Grunde stellst du dynamisch definierte Felder zur Verfügung, mit denen ein Benutzer die Props manipulieren kann, die den Komponenten deiner Stories übergeben werden. Folgendes werden wir entwickeln...
```
[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/using-addons.md]: Es sollte sein "herumexperimentieren" und "spielen" (nicht "herumspielen").

File: content/intro-to-storybook/react/de/using-addons.md [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+<strong>📝 Die Reihnfolge der Addon Registrierung ist wichtig!</strong>
+<br/>
+Die Reihenfolge, in der du die Addons auflistest, bestimmt die Reihenfolge, in der sie als Tabs im Addon-Panel erscheinen (für diejenigen, die dort angezeigt werden).
+</div>
+
+Das war's! Zeit, das Addon in der Story einzusetzen.
+
+### Verwendung
+
+Lass uns den Knob-Type "Objekt" in der `Task` Komponente benutzen.
```
[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/using-addons.md]: Ich behandle den "Knob-Type" als eine Art feststehender Begriff (so wie Story). Deutsch und Englisch zu vermischen, indem ich das erste Teilwort in Englisch und das Zweite in Deutsch schreibe, finde ich sehr befremdlich. Dann müsste es besser "Knopf-Typ" heißen, und da ich Knob nicht übersetzen wollte, übersetzte ich auch nicht Type.

Ich überlege, ob es evtl. Sinn macht, Knob mit Knopf zu übersetzen. Aber irgendwie fühlt sich das auch nicht richtig an. Wie ist deine Meinung zu dem Thema?

File: content/intro-to-storybook/react/de/using-addons.md [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+
+![Hier ist es in Storybook.](/intro-to-storybook/addon-knobs-demo-edge-case-in-storybook.png)
+
+Sofern wir [visuelle Regressions-Tests](/react/de/test/) verwenden, werden wir nun auch darauf aufmerksam gemacht, falls wir unsere Umbruch-Lösung je kaputt machen sollten. Solche versteckten Grenzfälle werden nur zu gerne vergessen.
+
+### Änderungen mergen
+
+Vergiss nicht, deine Änderungen in Git zu mergen!
+
+## Addons mit dem Team teilen
```
[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/using-addons.md]: Ja, im Sinne der Nachvollziehbarkeit würde ich aber von einer eigenständigen Verschiebung absehen.

File: content/intro-to-storybook/react/de/creating-addons.md [hunk 12]
```diff
@@ -0,0 +1,464 @@
...
+## Nächste Schritte
+
+Der nächste logische Schritt für unser Addon wäre, es in ein eigenes Paket umzuwandeln, um es dem gesamten Team zur Verfügung zu stellen und vielleicht sogar der ganzen Community.
+
+Aber das behandeln wir nicht in diesem Tutorial. Das Beispiel demonstriert, wie du die Storybook API nutzen kannst, um dein eigenes Addon zu schreiben und den Entwicklungs-Workflow weiter zu verbessern.
+
+So kannst du dein Addon noch weiter anpassen:
+
+- [Buttons zur Storybook hinzufügen](https://github.com/storybookjs/storybook/blob/next/addons/viewport/src/register.tsx#L8-L15)
+- [durch den Channel mit einem IFrame kommunizieren](https://github.com/storybookjs/storybook/blob/next/dev-kits/addon-roundtrip/README.md)
```
[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/creating-addons.md]: Es gibt nach meiner Recherche keine "richtige" deutsche Schreibweise für dieses Wort. Ich habe mal das F klein gemacht, aber es ist ein Subjektiv, daher schreibe ich es weiterhin groß (anstatt "technisch"). Spricht da aus deiner Sicht etwas dagegen?

File: content/intro-to-storybook/react/de/using-addons.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (jonathanherdt) [reply content/intro-to-storybook/react/de/using-addons.md]: Ich sehe kein Problem mit dem Mischen von Englisch und Deutsch, wir schreiben ja auch `Task`-Komponente 😊

File: content/intro-to-storybook/react/de/using-addons.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (jonathanherdt) [reply content/intro-to-storybook/react/de/using-addons.md]: Hm, aber wenn es so offensichtlich verkehrt ist, sollten wir es doch nicht reproduzieren.

File: content/intro-to-storybook/react/de/creating-addons.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (jonathanherdt) [reply content/intro-to-storybook/react/de/creating-addons.md]: Jap, meine Recherche hatte auch ergeben, dass es keine "richtige" Schreibweise gibt, nicht mal im Englischen (ich habe die Varianten `iFrame` und `iframe` gefunden). Ich wäre für ein kleines `i`, weil man das wenigstens von Apple-Produkten (iPhone, iMac) kennt.

File: content/intro-to-storybook/react/de/deploy.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+</p>
+</div>
+
+Schicke das Formular ab, um den Code auf dem `master` Branch von `taskbox` zu bauen und zu deployen.
+
+Wenn das erledigt ist, sehen wir eine Bestätigungs-Nachricht auf Netlify mit einem Link zum online verfügbaren `taskbox` Storybook. Hast du alle Anweisungen befolgt, sollte dein Storybook unter einer Subdomain wie [dieser hier](https://clever-banach-415c03.netlify.com/) zu erreichen sein.
+
+![Deploytes Storybook auf Netlify](/intro-to-storybook/netlify-storybook-deploy.png)
+
+Wir haben die Einrichtung von Continuous Deployment für unser Storybook abgeschlossen. Ab jetzt können wir unsere Stories mit den Mitgliedern in unserem Team per Link teilen.
```
[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/deploy.md]: Continuous Deployment ist ein feststehender Begriff, der nicht mit einem Bindestrich verbunden werden muss.

File: content/intro-to-storybook/react/de/deploy.md [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+
+Wenn du Storybook jetzt via `npm run build-storybook` baust, wird automatisch ein statisches Storybook im Verzeichnis `storybook-static` generiert.
+
+## Continuous Deployment
+
+Wir wollen die neueste Version unserer Komponenten verfügbar machen, wann immer wir Code pushen. Dazu müssen wir Continuous Deployment (CD) für Storybook einrichten. Wir nutzen dazu GitHub und Netlify, um unsere statische Seite zu deployen. Von Netlify nutzen wir dabei das kostenlose Paket.
+
+### GitHub
+
+Als Erstes musst du Git für dein lokales Projekt-Verzeichnis einrichten. Wenn du bereits das vorherige Kapitel "Tests" gelesen hast, springe zum Aufsetzen eines GitHub Repository.
```
[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/deploy.md]: Das Genitiv-s ist Ist bei nicht in der allgemeinen deutschen Sprache etablierten Fachbegriffen nicht vorgeschrieben.

File: content/intro-to-storybook/react/de/using-addons.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (renet) [reply content/intro-to-storybook/react/de/using-addons.md]: Ich übersetze nur, die Struktur (sowie der Inhalt der Beispiele, seien sie noch veraltet oder auch nicht), rühre ich nicht an, um Änderungen am Original zukünftig weiterhin ohne großen Aufwand übernehmen zu können.

**Comment by renet** [comment]
Thank you, @jonathanherdt, for taking the time to review my translation. Even though it was a lot of work adding all the dashes, I appreciate your feedback. ;)

I resolved all added comments (decided to ignore a very few of them, as they are a matter of opinion, though). Imho, this translation is now ready to get merged.

**Comment by domyen** [comment]
Awesome work René! @jonathanherdt give me the OK and I'll get this merged!

**Comment by jonathanherdt** [comment]
Good to go from my side!

File: content/intro-to-storybook/react/de/using-addons.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (jonathanherdt) [reply content/intro-to-storybook/react/de/using-addons.md]: Habe es in einem separaten Pull Request behandelt 😊 https://github.com/chromaui/learnstorybook.com/pull/262
